scrapy/pipelines/files.py

```diff
@@ -229,6 +229,19 @@ def __init__(self, uri):
         bucket, prefix = uri[5:].split('/', 1)
         self.bucket = client.bucket(bucket)
         self.prefix = prefix
+        permissions = self.bucket.test_iam_permissions(
+            ['storage.objects.get', 'storage.objects.create']
+        )
+        if 'storage.objects.get' not in permissions:
+            logger.warning(
+                "No 'storage.objects.get' permission for GSC bucket %(bucket)s",
```

Comment:
What do you think about adding more details about this problem - explaining that it would cause the file to be downloaded every time.

Reply:
Just pushed a commit. I hope it's fine now.